clang/test/dpct/helper_files_ref/include/util.hpp

```diff
@@ -620,6 +620,60 @@ class logical_group {
     return _group_linear_range_in_parent;
   }
 };
+/// Computes the active work-group number per Xe-Core when the occupancy is maximum.
+/// Ref to https://github.com/oneapi-src/oneAPI-samples/tree/master/Tools/GPU-Occupancy-Calculator
+/// \param [out] num_wg Active work-group number.
+/// \param [in] wg_size Work-group size.
+/// \param [in] slm_size Shared local memory size.
+/// \param [in] sg_size Sub-group size.
+/// \param [in] used_barrier Whether barrier is used.
+/// \param [in] used_large_grf Whether General Register File is used.
+/// \return Always return 0.
+inline int sycl_occupancy_max_active_wg_per_xecore(
+    int *num_wg, int wg_size, int slm_size = 0, int sg_size = 32,
+    bool used_barrier = false, bool used_large_grf = false) {
+  const int slm_size_per_xe_core = 64 * 1024;
+  const int max_barrier_registers = 32;
+  dpct::device_ext &dev = dpct::get_current_device();
+
+  size_t max_wg_size = dev.get_info<sycl::info::device::max_work_group_size>();
+  assert(wg_size <= max_wg_size && "Work-group size cannot exceed the limit.");
```

Comment:
let's limit wg_size to max_wg_size, have return value to tell caller.

Reply:
done